### PR TITLE
🐛 Fix bug: Fix pulling only files with tags

### PR DESCRIPTION
**Pull request by mrnicegyu11**
## What do these changes do?

- Fixes bug where deployment agent fails to poll. This happens for git-repos where only certain files are pulled and where a certain tag is regex-matched. Details for the fix are found in the comment in the code

Bonus:
- Clearer error when the console script to generate the stack.yml file fails
- make sure the cwd (`/tmp/blablabla`) is handed over to the script that generated the `stack.yml` file by using the env option of subprocess.pOpen

This has been tested by building and deploying the modifying deployment agent with these fixes in a local deploy and on aws-staging.

### CODE REVIEW

File: src/simcore_service_deployment_agent/auto_deploy_task.py [hunk 1]
```diff
@@ -158,6 +158,14 @@ async def generate_stack_file(app_config: Dict, git_task: GitUrlWatcher) -> Path
         raise ConfigurationError(
             "The stack file {} does not exist".format(stack_file.name)
         )
+    if (
+        stack_file.stat().st_size == 0
+    ):  # Filesize check via https://stackoverflow.com/a/55949699
+        raise ConfigurationError(
+            "The stack file {} is empty (0 bytes). Likely your console call to create it failed.".format(
```
Comment (sanderegg) [inline src/simcore_service_deployment_agent/auto_deploy_task.py:165]: suggestion what about refactoring both ifs in one go?
```python
if not stack_file.exists() or not stack_file.state().st_size:
    raise ConfigurationError(f"The generation of {stack_file=} failed using {stack_recipe_cfg['command']}")
```

**Review by Surfict — APPROVED** [review]

File: src/simcore_service_deployment_agent/git_url_watcher.py [hunk 2]
```diff
@@ -252,8 +252,17 @@ async def _init_repositories(repos: List[GitRepo]) -> Dict:
             raise ConfigurationError(
                 msg=f"no tags found in {repo.repo_url}:{repo.branch} that follows defined tags pattern {repo.tags}: {latest_tag}"
             )
-
+        # This next call was introcued to fix a bug. It is necessary since calls to *_checkout_repository*
+        # may only check out files at certain tags while HEAD stays at origin/master.
+        # If HEAD!=latest_tag, subsequent calls to *_git_get_current_matching_tag*
+        # will return an empty list since HEAD==origin/master is not tagged. This will make the deployment agent fail.
+        # I'd call this a workaround and a design deficiency (DK Nov2022)
```
Comment (sanderegg) [inline src/simcore_service_deployment_agent/git_url_watcher.py:259]: if possible add the link the github issue here

File: src/simcore_service_deployment_agent/git_url_watcher.py [hunk 3]
```diff
@@ -252,8 +252,17 @@ async def _init_repositories(repos: List[GitRepo]) -> Dict:
...
+        await _git_checkout_repo(repo.directory, latest_tag)
+        # This subsequent call, if repo.pull_only_files==true, will only checkout the specified files at the given revision
         await _checkout_repository(repo, latest_tag)
+        #
```
Comment (sanderegg) [inline src/simcore_service_deployment_agent/git_url_watcher.py:263]: unnecessary comments?

File: src/simcore_service_deployment_agent/cmd_utils.py [hunk 4]
```diff
@@ -38,7 +38,13 @@ async def run_cmd_line(cmd: List[str], cwd_: str = ".") -> str:
...
     proc = await asyncio.create_subprocess_shell(
-        cmd, stdout=asyncio.subprocess.PIPE, stderr=asyncio.subprocess.PIPE, cwd=cwd_
+        cmd,
+        stdout=asyncio.subprocess.PIPE,
+        stderr=asyncio.subprocess.PIPE,
+        cwd=cwd_,
+        env={
+            "PWD": cwd_,
+        },
     )
```
Comment (sanderegg) [inline src/simcore_service_deployment_agent/cmd_utils.py:48]: Question1: what is the idea behind having these 2 functions (run_cmd_line_unsafe vs run_cmd_line)? When do you use one or the other?
Question2: not sure I understand the fix here. so you override PWD by "." ? is this not supposed to be an absolute path? you mean that the ```cwd``` argument does not work? (if yes then what about the call above?) any additional information on that?

**Review by sanderegg — COMMENTED** [review]
Some questions on the fix. we can also discuss directly or tomorrow.

**Comment by mrnicegyu11** [comment]
yea let's talk tomorrow thanks for the input :)

File: src/simcore_service_deployment_agent/cmd_utils.py (hunk 4 above)
Comment (mrnicegyu11) [reply src/simcore_service_deployment_agent/cmd_utils.py:45]: 1: discussed in person
2: you are right, it is removed

File: src/simcore_service_deployment_agent/git_url_watcher.py (hunk 3 above)
Comment (mrnicegyu11) [reply src/simcore_service_deployment_agent/git_url_watcher.py:264]: linting for pleasant spacing ;)

File: src/simcore_service_deployment_agent/auto_deploy_task.py (hunk 1 above)
Comment (mrnicegyu11) [reply src/simcore_service_deployment_agent/auto_deploy_task.py]: done, note the typo on your suggestion: `state()` should be `stat()`

thanks a lot!

**Pull request merged**